feat: Rework Browser JS integration docs (#7648)
getsentry/sentry-docs: src/platforms/react-native/troubleshooting.mdx

- Checking that source maps are correctly uploaded with the exact same `release` and `dist` values as the event. They **must match** for source maps to upload correctly.
- If you set a custom `release` and `dist`, confirming that you manually uploaded the sourcemaps since our automatic source map upload script will not detect the custom values.
- Make sure that the name of the bundle in the stack trace matches exactly with the name in the source map. For example, if your stack frames say `app:///index.bundle`, the artifacts must be named `index.bundle` and `index.bundle.map` respectively.

> - Checking that you are not disabling the [`RewriteFrames`](/platforms/javascript/configuration/integrations/rewriteframes) default integration by overwriting it, filtering it out, or setting `defaultIntegrations: false`.

Reviewer: ```suggestion
- Checking that you're not disabling the [`RewriteFrames`](/platforms/javascript/configuration/integrations/rewriteframes) default integration by overwriting it, filtering it out, or setting `defaultIntegrations: false`.
```
